pkg/config/remote/client.go

```diff
@@ -15,46 +16,41 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/config/remote/data"
 	"github.com/DataDog/datadog-agent/pkg/config/remote/meta"
 	"github.com/DataDog/datadog-agent/pkg/proto/pbgo"
-	"github.com/DataDog/datadog-agent/pkg/remoteconfig/client"
-	agentgrpc "github.com/DataDog/datadog-agent/pkg/util/grpc"
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig"
+	"github.com/DataDog/datadog-agent/pkg/util/grpc"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
-	"github.com/DataDog/datadog-agent/pkg/version"
 	"github.com/pkg/errors"
-	"google.golang.org/grpc"
 	"google.golang.org/grpc/metadata"
 )
 
 // Client is a remote-configuration client to obtain configurations from the local API
 type Client struct {
-	m     sync.Mutex
-	ctx   context.Context
-	close func()
+	m sync.Mutex
+
+	ID string
+
+	startupSync sync.Once
+	ctx         context.Context
+	close       context.CancelFunc
 
 	agentName    string
+	agentVersion string
 	products     []string
-	pollInterval time.Duration
 
-	grpc           pbgo.AgentSecureClient
-	stateClient    *client.Client
-	currentConfigs client.Configs
+	pollInterval    time.Duration
+	lastUpdateError error
 
-	lastPollErr error
+	state *remoteconfig.Repository
 
-	apmSamplingUpdates chan []client.ConfigAPMSamling
-	cwsDDUpdates       chan []client.ConfigCWSDD
-}
+	grpc pbgo.AgentSecureClient
 
-// NewClient creates a new client
-func NewClient(agentName string, products []data.Product) (*Client, error) {
-	client, err := newClient(agentName, products)
-	if err != nil {
-		return nil, err
-	}
-	go client.pollLoop()
-	return client, nil
+	// Listeners
+	apmListeners []func(update map[string]remoteconfig.APMSamplingConfig)
+	cwsListeners []func(update map[string]remoteconfig.ConfigCWSDD)
 }
 
-func newClient(agentName string, products []data.Product, dialOpts ...grpc.DialOption) (*Client, error) {
+// NewClient creates a new client
+func NewClient(agentName string, agentVersion string, products []data.Product, pollInterval time.Duration) (*Client, error) {
 	token, err := security.FetchAuthToken()
```

Comment:
Note: I moved this to each poll query on main to mitigate the core-agent vs trace-agent startup race

Reply:
👍🏻  Will move it in this branch as well.